test/src/test/java/hudson/model/UserTest.java

```diff
@@ -783,4 +773,186 @@ public WebResponse getResponse(WebRequest request) throws IOException {
         assertThat(failingResources, empty());
     }
 
+    @Issue("JENKINS-55813")
+    private static class DisabledAccountsRealm extends AbstractPasswordBasedSecurityRealm {
+        private static class UserStub implements UserDetails {
```

Comment:
You do not need this stub. Just use `org.acegisecurity.userdetails.User`. That will simplify the merge with #4848.